### DISCUSSION TITLE
TS100 Temp sensor off by 500°C

### DISCUSSION THREAD

**Discussion by jorislue** [Ralim/IronOS#1031] [Q&A]
Hey, i tried several times to re-flash and to calibrate the temperature sensor, but on the current TS100 2.15 Build and also with 2.14 i get the following screen:
![20210826_182338](https://user-images.githubusercontent.com/2420996/131000103-9dc7e895-dd03-4c57-9ff4-175d013cf7e3.jpg)

calibrating the temp in the advanced settings didn't change anything :/
It works fine with the standard firmware. Connected /w 19V DC Power Supply

Edit: i just checked on the firmware: the temp sensor states -50°C when on room temp

**Comment by Ralim** [comment]
So if your ambient temperature sensor is reading -50C; then either that sensor is not working _or_ miniware have populated a different part for it. If the original firmware is also reading wrong I would strongly suggest you reach out for a replacement unit.

**Reply by jorislue** [reply]
I have already contacted them for a replacement, since I just ordered it a few days ago.
Nevertheless the difference between the IronOS and Miniware readings are odd and it would be interesting to find out if minware changed something or if it's just an issue with my hardware. I posted a pic with removed back cover in #1032

**Reply by Ralim** [reply]
Miniware constraings the value in a _lot_ more than IronOS.
Tradeoff : smaller range == better failure mode, wider range == better accuracy in wider operating temps

Looking at your image, indeed U6 is marked as a TMP36, so that matches all of the existing code and assumptions.

To rule it out, it might be worth popping the little daughter board off, cleaning the contact and re-inserting it.
I have had _odd_ failure modes when that connector is not correctly mated.

**Reply by jorislue** [reply]
Popped it out, cleaned with some isoprop and popped it back in. Observated the following:
When first launching on miniware OS, the Temp sensor stated 14° (without the tip). Put the tip back in: -42°C
Then popped the tip out: +428°C. Flashed IronOS on it again -> Same issue. So i guess i'll put everything together and send it back where it came from :P 
Thank you for your work and effort here tho - really appreciate it 👍

**Reply by Ralim** [reply]
Damn,
Worth a try just to rule it out.
I would advise having it replaced if you can.

If for some reason you are rejected post here and we can try and figure out what part is dody.
